packages/numpy/test_numpy.py

```diff
@@ -191,3 +194,115 @@ def test_runwebworker_numpy(selenium_standalone):
         """
     )
     assert output == "[0. 0. 0. 0. 0.]"
+
+
+def test_get_buffer(selenium):
+    selenium.run_js(
+        """
+        await pyodide.runPythonAsync(`
+            import numpy as np
+            x = np.arange(24)
+            z1 = x.reshape([8,3])
+            z2 = z1[-1::-1]
+            z3 = z1[::,-1::-1]
+            z4 = z1[-1::-1,-1::-1]
+        `);
+        for(let x of ["z1", "z2", "z3", "z4"]){
+            let z = pyodide.pyimport(x).getBuffer("u32");
+            for(let idx1 = 0; idx1 < 8; idx1++) {
+                for(let idx2 = 0; idx2 < 3; idx2++){
+                    let v1 = z.data[z.offset + z.strides[0] * idx1 + z.strides[1] * idx2];
+                    let v2 = pyodide.runPython(`repr(${x}[${idx1}, ${idx2}])`);
+                    console.log(`${v1}, ${typeof(v1)}, ${v2}, ${typeof(v2)}, ${v1===v2}`);
+                    if(v1.toString() !== v2){
+                        throw new Error(`Discrepancy ${x}[${idx1}, ${idx2}]: ${v1} != ${v2}`);
+                    }
+                }
+            }
+            z.release();
+        }
+        """
+    )
+
+
+@pytest.mark.parametrize(
+    "arg",
+    [
+        "np.arange(6).reshape((2, -1))",
+        pytest.param(
+            "np.arange(12).reshape((3, -1))[::2, ::2]", marks=pytest.mark.xfail
+        ),
```

Comment:
The buffer size (and contents) are different in this case. It's likely expected given this implementation where start and endoffsets are recomputed but I wonder if it is an issue.

Reply:
Yeah this is the expected behavior: if the buffer isn't contiguous the `data` field will contain extra entries that are not part of the array.
What does the `-1` in the shape mean?

Reply:
-1 in reshape means the remaining shape. So for instance `np.arange(6).reshape((2, -1))` is identical to `np.arange(6).reshape((2, 3))`. It will error if the remaining size isn't an int.

Reply:
OK, maybe we could keep this line with a comment but remove the xfail

Reply:
I will try to modify the test to assert the right thing for noncontiguous buffers.